File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/constant/Constant2Add.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Int32 import Int32

@PythonTaskInfo
class Constant2Add(PythonTask):

    def __init_ports__(self):
        self.in_1 = InputPort(name="In 1", data_type=Int32)
        self.in_2 = InputPort(name="In 2", data_type=Int32)
        self.out = OutputPort(name="Out", data_type=Int32)

    def execute(self, grp):
        self.in_1.get_data(Int32)
        suma = Int32(self.in_1.get_data(Int32).values[0] + self.in_2.get_data(Int32).values[0])
        self.out.put_data(suma)

File: com.systemincloud.examples.tasks.rtask/src/test/r/constant/Constant2Add.R

library(R6)
library(sicrtask)

Constant2Add <- R6Class("Constant2Add",
  inherit = RTask,
  public = list(
    rtaskinfo = RTaskInfo$new(),
    in1 = NA,
    in2 = NA,
    out = NA,
    initialize_ports = function() {
      self$in1 <- InputPort$new(name = "in1", data_type = "Int32")
      self$in2 <- InputPort$new(name = "in2", data_type = "Int32")
      self$out <- OutputPort$new(name = "out", data_type = "Int32")
    },
    initialize = function() {
      self$initialize_ports()
    },
    execute = function(grp) {
	  sum = Int32$new(self$in1$get_data("Int32")$value + self$in2$get_data("Int32")$value)
	  self$out$put_data(sum)
	}
  ),
  private = list(
  )
)
